S3-Keploy/bucket/bucket.go

```diff
@@ -61,29 +62,35 @@ func (basics BucketBasics) CreateOneBucket(bucket string) (message string) {
 	return msg
 }
 
-func (basic BucketBasics) UploadFile(fileName string, bucketName string) (message string) {
+func (basics Basics) UploadFile(fileName string, bucketName string) (message string) {
 	file, err := os.Open(fileName)
 	if err != nil {
 		log.Printf("Couldn't open file %v to upload. Here's why: %v\n", fileName, err)
 		return "File " + fileName + " not uploaded"
-	} else {
-		defer file.Close()
-		_, err = basic.S3Client.PutObject(context.TODO(), &s3.PutObjectInput{
-			Bucket: aws.String(bucketName),
-			Key:    aws.String(fileName),
-			Body:   file,
-		})
+	}
+
+	defer func() {
+		err = file.Close()
 		if err != nil {
-			log.Printf("Couldn't upload file %v to %v:%v. Here's why: %v\n",
-				fileName, bucketName, fileName, err)
-			return "File " + fileName + " not uploaded"
+			log.Println(err)
 		}
-		return "File " + fileName + " uploaded successfully"
+	}()
+
```

Comment:
why logging error twice?

Reply:
let me correct this

Reply:
Here, err is logged once for file.close() in defer func() and then for basics.S3Client.PutObject() method. So its not the same error.